src/Protocol/ActivityPub/Delivery.php

```diff
@@ -48,8 +49,13 @@ public static function deliver(string $inbox): array
 		$serverfail = false;
 
 		foreach ($posts as $post) {
+			$owner = User::getOwnerDataById($post['uid']);
+			if (!$owner) {
+				continue;
```

Comment:
I think that before continuing here, we have to drop the post from the delivery queue by calling `Post\Delivery::remove(...);`

Reply:
When the result of the transmit call is `success => false, drop => true` (which is the equivalent of not having the owner record), the condition later in this loop specifically doesn’t call `Post\Delivery::remove` ?

Reply:
I hadn't got a deeper look, I must admit.

Reply:
Now I had a deeper look. `deliver` would return a "success" since the `$uri_ids` array was empty. But then the queue entry isn't removed. The removal of a successful transmission is done in `deliverToInbox`. There `Post\Delivery::incrementFailed(...);` is called as well.

Reply:
Both the remove and increment failed need to be called?

Reply:
Yeah, I think so.

Reply:
Done.